filament/src/details/Material.cpp

```diff
@@ -221,12 +221,12 @@ FMaterial::FMaterial(FEngine& engine, const Material::Builder& builder)
             mRasterState.depthWrite = true;
             break;
         case BlendingMode::MASKED:
-            // MASKED mode now leaves destination alpha intact.
-            // This prevents strange behavior with semi-transparent render targets, which the
-            // model viewer team discovered when testing against the Khronos alpha test
-            // conformance model.
+            // This leaves destination alpha intact if it is already opaque, but makes it opaque if
+            // it is already semi-transparent. This is usually what you want when drawing a scene.
+            // Do not change this behavior without checking for regressions with AlphaBlendModeTest
+            // and TextureLinearInterpolationTest, with and without View::BlendMode::TRANSLUCENT.
             mRasterState.blendFunctionSrcRGB   = BlendFunction::ONE;
-            mRasterState.blendFunctionSrcAlpha = BlendFunction::ZERO;
+            mRasterState.blendFunctionSrcAlpha = BlendFunction::SRC_ALPHA;
```

Comment:
If I'm not mistaken, with this setup the alpha value is calculated as:

`result_alpha = src_alpha * src_alpha + dst_alpha`

this just seems wrong, This can lead to alpha values > 1, and I don't think there is a justification for the src_alpha^2.

Why not use the standard equation: ONE / ONE_MINUS_SRC_ALPHA, which would lead to:

`result_alpha = src_alpha + (1 - src_alpha) * dst_alpha`
`result_alpha = dst_alpha + src_alpha * (1 - dst_alpha)`

The destination alpha can only increase, i.e. only get more opaque. If src_alpha is 1, then result is 1 and if dst_alplha is 1, then it stays to 1. I think this is what you wanted.

Reply:
alpha>1 doesn't matter since it will be clamped on output anyway. But yeah, `ONE_MINUS_SRC_ALPHA` is better for DstAlpha.

Reply:
I don't think it's clamped for float targets.